File: week4/Java/237. Delete Node in a Linked List.java
// since we do not have any information about the previous node, and we are only givin the node that we need to delete, we just need to copy the next node's address and the value into the current node.

class Solution {
    public void deleteNode(ListNode node) {
        node.val = node.next.val;
    node.next = node.next.next;
    }
}

File: week4/python/237. Delete Node in a Linked List.py
#since we do not have any information about the previous node, and we are only givin the node that we need to delete, we just need to copy the next node's address and the value into the current node.

class Solution:
    def deleteNode(self, node):
        """
        :type node: ListNode
        :rtype: void Do not return anything, modify node in-place instead.
        """
        if node.next is not None:
            node.val, node.next = node.next.val, node.next.next
